smithy-cli/src/main/java/software/amazon/smithy/cli/commands/ClasspathAction.java

```diff
@@ -133,13 +134,32 @@ private List<Path> resolveDependencies(
         long lastModified = smithyBuildConfig.getLastModifiedInMillis();
         DependencyResolver delegate = new FilterCliVersionResolver(SmithyCli.getVersion(), baseResolver);
         DependencyResolver resolver = new FileCacheResolver(getCacheFile(buildOptions, smithyBuildConfig),
-                                                            lastModified,
-                                                            delegate);
-        addConfiguredMavenRepos(smithyBuildConfig, resolver);
-        maven.getDependencies().forEach(resolver::addDependency);
+                lastModified,
+                delegate);
+
+        Set<MavenRepository> repositories = DependencyUtils.getConfiguredMavenRepos(smithyBuildConfig);
+        Set<String> dependencies = maven.getDependencies();
+
+        // Use the lock file dependencies and repositories if a lockfile exists
+        Optional<LockFile> lockFileOptional = DependencyUtils.loadLockfile();
+        if (lockFileOptional.isPresent()) {
+            if (lockFileOptional.get().getConfigHash() != DependencyUtils.configHash(dependencies, repositories)) {
+                throw new CliError(
+                        "Lockfile does not match configured dependencies. Re-lock dependencies or revert changes.");
```

Comment:
Can you say smithy-lock.json or whatever it's called rather than LockFile? And also say what command to use to re-lock dependencies?